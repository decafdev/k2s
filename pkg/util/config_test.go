package util

import (
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func TestNewConfig(t *testing.T) {
	// given struct
	type given struct {
		params []string
	}

	tests := make(map[string]given)

	tests["when no values are passed"] = given{
		params: []string{""},
	}

	tests["when defined parameters are provided"] = given{
		params: []string{
			"VERSION=99.99.99",
			"LOGGER_LEVEL=DEBUG",
			"LOG_PRETTY_PRINT=TRUE",
			"ENVIRONMENT=PROD",
			"PORT=1337",
		},
	}

	tests["when validation fails"] = given{
		params: []string{
			"VERSION=not-semver",
			"LOGGER_LEVEL=invalid",
			"LOG_PRETTY_PRINT=invalid",
			"PORT=should-be-numeric",
		},
	}

	for when, given := range tests {
		t.Run(when, func(t *testing.T) {
			svc, err := NewConfig(given.params...).Validate()

			if err != nil {
				snaps.MatchSnapshot(t, svc, err.Error())
			} else {
				snaps.MatchSnapshot(t, svc, nil)
			}
		})
	}
}
