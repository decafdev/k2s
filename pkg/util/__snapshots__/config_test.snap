
[TestNewConfig/when_no_values_are_passed - 1]
&util.Config{SERVICE_NAME:"", VERSION:"0.0.0", BASE_PATH:"/", PORT:"3000", ENVIRONMENT:"local", LOGGER_PRETTY_PRINT:"false", LOGGER_LEVEL:"info", TRAEFIK_VERSION:"2.5.4", TRAEFIK_REPLICAS:"1", PRIVATE_REGISTRY_ENABLED:"false", PRIVATE_REGISTRY_URL:"https://index.docker.io/v1/", PRIVATE_REGISTRY_USER:"", PRIVATE_REGISTRY_PASS:""}
nil
---

[TestNewConfig/when_defined_parameters_are_provided - 1]
&util.Config{SERVICE_NAME:"", VERSION:"99.99.99", BASE_PATH:"/", PORT:"1337", ENVIRONMENT:"prod", LOGGER_PRETTY_PRINT:"false", LOGGER_LEVEL:"debug", TRAEFIK_VERSION:"2.5.4", TRAEFIK_REPLICAS:"1", PRIVATE_REGISTRY_ENABLED:"false", PRIVATE_REGISTRY_URL:"https://index.docker.io/v1/", PRIVATE_REGISTRY_USER:"", PRIVATE_REGISTRY_PASS:""}
nil
---

[TestNewConfig/when_validation_fails - 1]
&util.Config{SERVICE_NAME:"", VERSION:"not-semver", BASE_PATH:"/", PORT:"should-be-numeric", ENVIRONMENT:"local", LOGGER_PRETTY_PRINT:"false", LOGGER_LEVEL:"invalid", TRAEFIK_VERSION:"2.5.4", TRAEFIK_REPLICAS:"1", PRIVATE_REGISTRY_ENABLED:"false", PRIVATE_REGISTRY_URL:"https://index.docker.io/v1/", PRIVATE_REGISTRY_USER:"", PRIVATE_REGISTRY_PASS:""}
Key: 'Config.VERSION' Error:Field validation for 'VERSION' failed on the 'semver' tag
Key: 'Config.PORT' Error:Field validation for 'PORT' failed on the 'numeric' tag
Key: 'Config.LOGGER_LEVEL' Error:Field validation for 'LOGGER_LEVEL' failed on the 'oneof' tag
---
